_pages/tutorials/programming/python/getting-started.md

```diff
@@ -0,0 +1,117 @@
+
+```python
+i = 10
+print i
+```
+
+If you're using a Mac, run your code by opening a program called `terminal` or if you're on windows open the `start menu` and type `cmd` and hit enter. Once open, type `cd ~/Desktop` and hit enter. Then type `python countdown.py` and hit enter again. If everything worked correctly, you should be seeing the value your program is printing which is the value we set `i` to earlier. Pretty cool right? To run your code again, just type `python countdown.py` again and press enter. Now that we have a basic understanding of methods, we can move on to making it do more than just printing.
+
+## Step 3: Looping
+
+Now were going to look at `while` loops. A `while` loop is basically just a way for a piece of your code to run over and over again. Here's how you add it to your code:
+
+```python
+i = 10
+while True:
+    print i
+```
+
+Now if you run this code, it should just print `10` and keep going forever. To stop this code, press `Ctrl-c`. Also, it it important to note that the indentation in the above code is important as it is how python identifies which parts of the code are affected by the while loop. Now that we have a loop, let's change the variable.
+
+## Step 4: Changing the Variable
+
+To modify the variable during the while loop, add this to your code:
+
+```python
+i = 10
+while True:
+    print 10 - i
+    i = i - 1
+```
+
+What this does is it takes what you had stored as `i` earlier and every time it runs it takes `i`, subtracts 1, and then sets that value to be `i`. Now if you run your code it should countdown put keep going forever. To fix this, we need to add a condition to make the loop stop.
+
+## Step 5: Stopping the loop
+
+To stop the loop, we need to add something to the while loop to tell it when to finish. we can do this by writing:
+
+```python
+i = 10
+while i > 0:
+    print 10 - i
+    i = i - 1
+```
+
+This will mean that the loop will stop running once `i` is less that 0. Now if you run your program, it should countdown from 10 to 1. But it is still too fast to be an actual countdown, so to fix this, we need to add a library.
+
+## Step 6: Libraries
+
+What are libraries? Well if you think about methods as a tool, then a library is like a toolbox. A library is essentially where many methods are stored and can be called by your code to perform different tasks. To add a library to your code, you first need to `import` it. The library we are going to use is called the time library. We can get this library by adding `import time` to the very top of our program. Once this is done we are going to use the `Sleep` method from this library. What this library does is it waits or "sleeps" for the amount of time you tell it to. We can add the sleep method to our code in the `while` loop by writing `time.Sleep(1)` which is pretty much telling our code "from the `time` library, use the `Sleep` method and tell it to wait 1 second." After this has been added to your code, it should look like this:
```

Comment:
@andrewda Nah, this is what I was talking about. This line is like 1000 chars long or something.